libsplinter/src/node_registry/mod.rs

```diff
@@ -28,24 +28,65 @@ pub struct Node {
     pub metadata: HashMap<String, String>,
 }
 
+/// A predicate on a key/value pair in a Node's metadata table.
+///
+/// Each variant is an operator, and supplies a tuple representing a key/ value pair. It is applied
+/// by the comparison operator on the valued found at the given key (the first item in the tuple)
+/// against the predicate's value  (the second item in the tuple).
+///
+/// If the item is missing in a node's metadata table, the predicate returns false.
+pub enum MetadataPredicate {
+    /// Applies the `==` operator for
```

Comment:
This comment should probably be removed

Reply:
Forgot to actually complete the doc strings for those.  Corrected.